python/pyspark/mllib/clustering.py

```diff
@@ -38,13 +38,175 @@
 
 
 @inherit_doc
+class BisectingKMeansModel(JavaModelWrapper):
+    """
+    .. note:: Experimental
+
+    A clustering model derived from the bisecting k-means method.
+
+    >>> data = array([0.0,0.0, 1.0,1.0, 9.0,8.0, 8.0,9.0]).reshape(4, 2)
+    >>> bskm = BisectingKMeans()
+    >>> model = bskm.run(sc.parallelize(data))
+    >>> model.predict(array([0.0, 0.0])) == model.predict(array([0.0, 0.0]))
+    True
+    >>> model.k
+    4
+    >>> model.computeCost(array([0.0, 0.0]))
+    0.0
+    >>> model.k == len(model.clusterCenters)
+    True
+    >>> model = bskm.setK(2).run(sc.parallelize(data))
+    >>> model.predict(array([0.0, 0.0])) == model.predict(array([1.0, 1.0]))
+    True
+    >>> model.k
+    2
+
+    .. versionadded:: 1.6.0
+    """
+
+    @property
+    @since('1.6.0')
+    def clusterCenters(self):
+        """Get the cluster centers, represented as a list of NumPy arrays."""
+        return [c.toArray() for c in self.call("clusterCenters")]
+
+    @property
+    @since('1.6.0')
+    def k(self):
+        """Get the number of clusters"""
+        return self.call("k")
+
+    @since('1.6.0')
+    def predict(self, x):
+        """
+        Find the cluster to which x belongs in this model.
+
+        :param x: Either the point to determine the cluster for or an RDD of points to determine
+        the clusters for.
+        """
+        if isinstance(x, RDD):
+            return x.map(self.predict(x))
+
+        x = _convert_to_vector(x)
+        return self.call("predict", x)
+
+    @since('1.6.0')
+    def computeCost(self, point):
+        """
+        Return the Bisecting K-means cost (sum of squared distances of points to
+        their nearest center) for this model on the given data.
+
+        :param point: the point to compute the cost to
+        """
+        return self.call("computeCost", _convert_to_vector(point))
+
+
+class BisectingKMeans:
+    """
+    A bisecting k-means algorithm based on the paper "A comparison of document clustering
+    techniques" by Steinbach, Karypis, and Kumar, with modification to fit Spark.
+    The algorithm starts from a single cluster that contains all points.
+    Iteratively it finds divisible clusters on the bottom level and bisects each of them using
+    k-means, until there are `k` leaf clusters in total or no leaf clusters are divisible.
+    The bisecting steps of clusters on the same level are grouped together to increase parallelism.
+    If bisecting all divisible clusters on the bottom level would result more than `k` leaf
+    clusters, larger clusters get higher priority.
+
+    Based on [[http://glaros.dtc.umn.edu/gkhome/fetch/papers/docclusterKDDTMW00.pdf
+    Steinbach, Karypis, and Kumar, A comparison of document clustering techniques,
+    KDD Workshop on Text Mining, 2000.]]
+
+    .. versionadded:: 1.6.0
+    """
+    def __init__(self):
+        self.k = 4
+        self.maxIterations = 20
+        self.minDivisibleClusterSize = 1.0
+        self.seed = 42
```

Comment:
I found the default value of ```seed``` is ```classOf[BisectingKMeans].getName.##``` at Scala side, I think we should keep consistent.